src/plugins/jqplot.enhancedLegendRenderer.js

```diff
@@ -267,24 +268,28 @@
             }
 
             else {
+                sidx1 = sidx;
+                if(reverse) {
+                    sidx1 = plot.series.length - 1 - sidx;
```

Comment:
I think we should colcurate the index of the array instead of rearranging the array.